provider/internal/msp/msp_tenant_user_api_token/resource.go

```diff
@@ -0,0 +1,114 @@
+import (
+	"context"
+	"fmt"
+	cdoClient "github.com/CiscoDevnet/terraform-provider-cdo/go-client"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/msp/users"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+func NewMspManagedTenantUserApiTokenResource() resource.Resource {
+	return &MspManagedTenantUserApiTokenResource{}
+}
+
+type MspManagedTenantUserApiTokenResource struct {
+	client *cdoClient.Client
+}
+
+func (m *MspManagedTenantUserApiTokenResource) Metadata(ctx context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = request.ProviderTypeName + "_msp_managed_tenant_user_api_token"
+}
+
+func (m *MspManagedTenantUserApiTokenResource) Schema(ctx context.Context, request resource.SchemaRequest, response *resource.SchemaResponse) {
+	response.Schema = schema.Schema{
+		MarkdownDescription: "Provides a resource to generate an API token for a user in an MSP-managed tenant.",
+		Attributes: map[string]schema.Attribute{
+			"tenant_uid": schema.StringAttribute{
+				MarkdownDescription: "Universally unique identifier of the tenant in which the API token for the user should be generated.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"user_uid": schema.StringAttribute{
+				MarkdownDescription: "Universally unique identifier of the user for whom the API token should be generated.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"api_token": schema.StringAttribute{
+				MarkdownDescription: "The generated API token for the user.",
+				Computed:            true,
+				Sensitive:           true,
+			},
+		},
+	}
+}
+
+func (m *MspManagedTenantUserApiTokenResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var planData MspManagedTenantUserApiTokenResourceModel
+	response.Diagnostics.Append(request.Plan.Get(ctx, &planData)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	tflog.Debug(ctx, fmt.Sprintf("Generating an API token for a user %s in the tenant %s...", planData.UserUid, planData.TenantUid))
+
+	apiTokenInfo, err := m.client.GenerateApiTokenForUserInMspManagedTenant(ctx, users.MspGenerateApiTokenInput{
+		UserUid:   planData.UserUid.ValueString(),
+		TenantUid: planData.TenantUid.ValueString(),
+	})
+	if err != nil {
+		response.Diagnostics.AddError(fmt.Sprintf("failed to generate API token for user %s in MSP-managed tenant %s", planData.UserUid, planData.TenantUid), err.Error())
```

Comment:
The nittest of nits: I think you may want to uppercase "failed"

Reply:
Good catch, Sir, thanks fixed.